### PR TITLE
Update sort-code validation

### PR DESCRIPTION
**Pull request by elenagarrone**
# Description

Update sort code validation as it wasn't working correctly and add notes on how to make the validation fully work when using the component. The notes have also been updated for the `date-field`.

Also add unsubscription for existing subscriptions.

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [ ] I have provided a story in storybook to document the changes
- [x] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: projects/canopy/src/lib/forms/date/date-field.component.ts [hunk 1]
```diff
@@ -144,30 +144,30 @@ export class LgDateFieldComponent implements OnInit, ControlValueAccessor, OnDes
       this.ngControl.control.updateValueAndValidity();
     }
 
-    this.valueChanges = this.dateFormGroup.valueChanges.subscribe((date: DateField) => {
-      const day = date.date ? ('0' + date.date).slice(-2) : '';
-      const month = date.month ? ('0' + date.month).slice(-2) : '';
-      const year = date.year ? date.year : '';
-      this.onChange(`${year}-${month}-${day}`);
-    });
+    this.subscriptions.push(
```
Comment (elenagarrone) [inline projects/canopy/src/lib/forms/date/date-field.component.ts:147]: Andy suggested to write tests for this but I honestly don't know how to do it? I don't really want to assign variables to the subscriptions only for testing purposes. We don't test unsubscriptions anywhere else and I'm not 100% sure I see the benefit.

Comment (owensgit) [reply projects/canopy/src/lib/forms/date/date-field.component.ts:147]: I'd say maybe something to keep an eye on, I think there are ways to do it, but wouldn't hold this PR up for it 👍

File: projects/canopy/src/lib/forms/date/date-field.notes.ts [hunk 2]
```diff
@@ -17,7 +17,8 @@ Import the Date Input Module into your application:
 and in your HTML:
 
 ~~~html
-<lg-date-field formControlName="date">
+<form [formGroup]="form" (ngSubmit)="..." #validationForm="ngForm">
```
Comment (owensgit) [inline projects/canopy/src/lib/forms/date/date-field.notes.ts:20]: I wonder if we could add a HTML comment as well, to make validation thing easy to notice at a glance. E.g.

```suggestion
<!-- It's important to include \`ngForm\` on the form tag for validation to work properly  -->
<form [formGroup]="form" (ngSubmit)="..." #validationForm="ngForm">
```

What do you think?

File: projects/canopy/src/lib/forms/date/date-field.notes.ts [hunk 3]
```diff
@@ -46,6 +47,8 @@ and in your HTML:
   </ng-container>
 </lg-validation>
 </lg-date-field>
+  <button type="submit">Submit</button>
```
Comment (owensgit) [inline projects/canopy/src/lib/forms/date/date-field.notes.ts:50]: Similar to my previous comment, not super fussed either way.

```suggestion
  <!-- submit button must be inside the form tag for validation to work properly -->
  <button type="submit">Submit</button>
```

File: projects/canopy/src/lib/forms/sort-code/sort-code.component.spec.ts [hunk 4]
```diff
@@ -258,6 +258,19 @@ describe('SortCodeComponent', () => {
...
+      sortCodeFieldInstance.second.markAsDirty();
+      sortCodeFieldInstance.third.markAsDirty();
+      sortCodeFieldInstance.first.setValue('xx');
+      sortCodeFieldInstance.second.setValue('');
+      sortCodeFieldInstance.third.setValue('xx');
+      fixture.detectChanges();
+      expect(fixture.componentInstance.form.controls.sortCode.errors).toEqual({
+        invalidField: true,
+      });
+    });
```
Comment (owensgit) [inline projects/canopy/src/lib/forms/sort-code/sort-code.component.spec.ts:272]: Nice one 🙌

**Review by owensgit — COMMENTED** [review]
Nice one @elenagarrone 👍  Added a couple of small questions.

File: projects/canopy/src/lib/forms/date/date-field.notes.ts (hunk 2 above)
Comment (elenagarrone) [reply projects/canopy/src/lib/forms/date/date-field.notes.ts:20]: Yes, I'll copy the entire comment in there too. The submit button within the form tag is also equally important so it's worth adding that too 👍

Comment (elenagarrone) [reply projects/canopy/src/lib/forms/date/date-field.notes.ts:20]: @owensgit the comments get removed when deploying the code so. I've added a bold note under the html.

File: projects/canopy/src/lib/forms/date/date-field.notes.ts [hunk 5]
```diff
@@ -17,37 +17,43 @@ Import the Date Input Module into your application:
...
+        Enter a date of birth
+      </ng-container>
+    </lg-validation>
+  </lg-date-field>
+
+  <button type="submit">Submit</button>
+</form>
 ~~~
 
+**Note: for the validation to work properly it's very important to include the \`ngForm\` on the form tag and for the submit button to be within the form.**
```
Comment (elenagarrone) [inline projects/canopy/src/lib/forms/date/date-field.notes.ts:55]: I know we now have two places in the same notes where we repeat this sentence but i'm keen to leave both of them as it's such an important step.

Comment (owensgit) [reply projects/canopy/src/lib/forms/date/date-field.notes.ts:55]: I agree 👍

**Review by owensgit — APPROVED** [review]

**Review by paul-request — APPROVED** [review]

**Pull request merged**